apps/speed.c

```diff
@@ -2011,14 +2011,17 @@ int speed_main(int argc, char **argv)
         case OPT_ASYNCJOBS:
 #ifndef OPENSSL_NO_ASYNC
             async_jobs = opt_int_arg();
+            if (async_jobs > 99999) {
+                BIO_printf(bio_err, "%s: too many async_jobs\n", prog);
+                goto opterr;
+            }
             if (!ASYNC_is_capable()) {
                 BIO_printf(bio_err,
                            "%s: async_jobs specified but async not supported\n",
                            prog);
-                goto opterr;
-            }
-            if (async_jobs > 99999) {
-                BIO_printf(bio_err, "%s: too many async_jobs\n", prog);
+                if (testmode)
+                    /* Return success in the testmode. */
+                    return 0;
```

Comment:
silent success here doesn't make sense to me.  Just because ASYNC_is_capable returns 0, doesn't mean speed is incapable of functioning, it just means its only capable of running a single job at a time.  i.e. run time failure in getcontext is a problem, but its not fatal, the test can still continue as a non-async execution.

It seems like instead quietly doing nothing in test mode, we should issue a warning about async not being supported and reduce the async_jobs value to 0.

Reply:
We extensively test speed in non-async mode. The point about the test which uses "-async_jobs" is to test the async jobs specific code which it cannot do if there is no async support. So it seems perfectly reasonable to me finish here.

Reply:
ok

Reply:
The testmode is just for testing the speed command as of the test recipe, it will not do anything particularly useful apart from that and other (non -async_jobs) recipes exist anyway. There is no point in continuing.